src/DDexport.cpp

```diff
@@ -491,28 +350,26 @@ namespace dd {
                     next_index++;
 
                     if (writeBinary) {
-                        os.write(reinterpret_cast<const char*>(&node_index[node->p]), sizeof(int));
-                        os.write(reinterpret_cast<const char*>(&node->p->v), sizeof(short));
+                        os.write(reinterpret_cast<const char*>(&node_index[node->p]), sizeof(std::int_fast64_t));
```

Comment:
We could write something like `sizeof(decltype(node_index[node->p]))` to avoid repeating the type. Would apply to the writes above and below as well.

Reply:
That should most certainly minimise error potential 👍🏼